Update release issue templates regarding release documentation (#7498)
medic/cht-core: .github/ISSUE_TEMPLATE/z_release_major.md

```
- [ ] Build a beta named `<major>.<minor>.<patch>-beta.1` by pushing a git tag and when CI completes successfully notify the QA team that it's ready for release testing.
- [ ] Add release notes to the [Core Framework Releases](https://docs.communityhealthtoolkit.org/core/releases/) page:
  - [ ] Create a new document for the release in the [releases folder](https://github.com/medic/cht-docs/tree/main/content/en/core/releases). 
  - [ ] Ensure all issues are in the GH Milestone, that they're correctly labelled (in particular: they have the right Type, "UI/UX" if they change the UI, and "Breaking change" if appropriate), and have human readable descriptions. 
  - [ ] Use [this script](https://github.com/medic/cht-core/blob/master/scripts/release-notes) to export the issues into our release note format. 
  - [ ] Manually document any known migration steps and known issues. 
  - [ ] Provide description, screenshots, videos, and anything else to help communicate particularly important changes. 

>   - [ ] Document any required or recommended upgrades to our other products (eg: medic-conf,  medic-gateway, medic-android).

Reviewer: ```suggestion
  - [ ] Document any required or recommended upgrades to our other products (eg: cht-conf,  cht-gateway, cht-android, etc).
```

  - [ ] Add the release to the [Supported versions](https://docs.communityhealthtoolkit.org/core/releases/) and update the EOL date and status of previous releases. Also add a link in the `Release Notes` section to the new release page.
  - [ ] Assign the PR to:
    - The Director of Technology
    - An SRE to review and confirm the documentation on upgrade instructions and breaking changes is sufficient
- [ ] Until release testing passes, make sure regressions are fixed in `master`, cherry-pick them into the release branch, and release another beta.
- [ ] Create a release in GitHub from the release branch so it shows up under the [Releases tab](https://github.com/medic/cht-core/releases) with the naming convention `<major>.<minor>.<patch>`. This will create the git tag automatically. Link to the release notes in the description of the release.
- [ ] Confirm the release build completes successfully and the new release is available on the [market](https://staging.dev.medicmobile.org/builds/releases). Make sure that the document has new entry with `id: medic:medic:<major>.<minor>.<patch>`

> - [ ] Review the scalability results on S3 at medic-e2e/scalability/$TAG_NAME. Add the release `.jtl` file to `cht-core/tests/scalability/previous_results`. Compare the trend using the [scalability documentation](https://github.com/medic/cht-core/blob/master/tests/scalability/README.md).

Author: @garethbowen Does it make more sense to move this up and do it before actually cutting the release (as a part of the beta tag steps)?

Just seems like if we do see a performance issue here, we are too late since the release is already out....

Reviewer: Yeah I think this is poorly worded. Analysing the scalability results should be part of the release testing process. Only the final build results should be copied into the repo (the betas can be discarded).

Please confirm with QA that there is a documented step in their release testing process for evaluating the scalability and performance testing results, and then reword this to something like...

```suggestion
- [ ] Execute the scalability testing suite on the final build and download the scalability results on S3 at medic-e2e/scalability/$TAG_NAME. Add the release `.jtl` file to `cht-core/tests/scalability/previous_results`. More info in the  [scalability documentation](https://github.com/medic/cht-core/blob/master/tests/scalability/README.md).
```

Reviewer: Going with this updated wording. Bede is currently automating more of this so it'll change again soon but don't want to hold up these changes.

- [ ] Upgrade the `demo-cht.dev` instance to this version.
- [ ] Announce the release on the [CHT forum](https://forum.communityhealthtoolkit.org/c/product/releases/26), under the "Product - Releases" category using this template:
```
*We're excited to announce the release of {{version}} of {{product}}*

New features include {{key_features}}. We've also implemented loads of other improvements and fixed a heap of bugs.

> Read the [release notes]({{url}}) for full details.

Author: These changes to the release announcement were just minor things I noticed during the `3.13` release. Seems like there is no need to show the raw URLs when we can use the markdown formatting...

But let me know if I should leave this how it was or if there are other changes you think we should make here.

Reviewer: Nice improvement. I think this evolved from when we used to post the message to Slack manually. Now that a bot updates Slack it's not needed. Good spotting!

Following our support policy, versions {{versions}} are no longer supported. Projects running these versions should start planning to upgrade in the near future. For more details read our [software support documentation](https://docs.communityhealthtoolkit.org/core/overview/supported-software/).

> Check our [milesones](https://github.com/medic/cht-core/milestones) to see what's scheduled for the next releases.

Reviewer: I think this should probably point to the roadmap instead, or as well, as the milestones...

```suggestion
Check out our [roadmap](https://github.com/orgs/medic/projects/112) to see what we're working on next.
```
